### DISCUSSION TITLE
A couple of issues trying to upgrade to current version of Vuepress and theme hope

### DISCUSSION THREAD

**Discussion by Daeraxa** [vuepress-theme-hope/vuepress-theme-hope#2966] [Q&A]
I'm currently trying to get our website (https://pulsar-edit.dev) updated to a more current version of vuepress and theme hope. All the changes I've made so far to get this working can be found here:
https://github.com/pulsar-edit/pulsar-edit.github.io/pull/198/files

There are a lot of changes it seems we have to make due to changes such as:
- `@include` syntax change
- Adding `include: true` to mdEnhance
- Outdated sidebar terms (`title`, `collapsable`->`collapsible`)
- Outdated config.js terms (`editLinks->editLink`, `displayAllHeaders`)

The above are all fine and I can manage those easily. Where I'm getting stuck is on a few other areas mainly:

## Fontawesome icons
None of the icons show but it seems the syntax has changed? For example if I amend `solid file-lines` to `fas file-lines` then it works (as it seems do all the other `solid` type). However I cannot get `brands` to work e.g. `brands fa-discord` -> `fab fa-discord`. (`brands` are still in the free set).

## Custom `@include` filepath (https://theme-hope.vuejs.press/guide/markdown/include.html#advanced)
It seems I've had to update from `getPath` to `resolvePath` according to the docs but I still can't get this to work, probably me being stupid...
This was working before:
```js
        include: {
          getPath: (file) => {
            if (file.startsWith("@orgdocs")) {
              return file.replace("@orgdocs",
                path.resolve(__dirname, "../../node_modules/.github/"));
            }
            return file;
          },
```
but this is what I've updated to and can't get working:
```js
        include: [
          true,
          {
          resolvePath: (file) => {
            if (file.startsWith("@orgdocs")) {
              return file.replace("@orgdocs",
                path.resolve(__dirname, "../../node_modules/.github/"));
            }
            return file;
          },
        }],
```
Syntax on the page's `index.md` is updated from:
`@include(@orgdocs/PRIVACY_POLICY.md)`
to
`<!-- include: @orgdocs/PRIVACY_POLICY.md -->` (also tried `<!-- include: ./@orgdocs/PRIVACY_POLICY.md -->`

## Custom frontpage section not displaying

One of our community members created a new frontpage section and hero which are no longer showing at all, not sure if this is just a Vuepress upgrade thing or some kind of change in the theme:

This is difficult to explain by discussion so you can see the PR that added this feature here: https://github.com/pulsar-edit/pulsar-edit.github.io/pull/144/files

Similarly it seems a script no longer runs as it did before: https://github.com/pulsar-edit/pulsar-edit.github.io/pull/139/files

No changes were made to either of the above.

I know this is a long post but any help would be most appreciated here.

**Comment by Mister-Hope** [comment]
I am on the train home, will reply later.

Fontawesome brands requires a longer keyword, check that.

**Reply by Daeraxa** [reply]
No problem, appreciate your help.
As for the FA brands, not sure what it should be then as all I can find is either `fab` or `fa-brands` and neither seem to work.

**Reply by Mister-Hope** [reply]
I mean fontawesome-with-brand keyword, not sure if it ends with `s`, check the docs.

**Reply by Mister-Hope** [reply]
I am not sure why you update include like that? Just rename the option would be fine.

**Reply by Mister-Hope** [reply]
The alias no longer ends with `.js`, check the docs, some webpack loader would check the extension, and assigning a js file to another file as alias can be problematic in some cases.

**Reply by Daeraxa** [reply]
Thanks for the reply and help.

> I mean fontawesome-with-brand keyword, not sure if it ends with `s`, check the docs.

Ok the other keyword did work, I must have missed that in the docs because it worked with just `fontawesome` in the previous version. So that is one issue crossed off.

> I am not sure why you update include like that? Just rename the option would be fine.

Sorry, I'm not quite sure what you mean. If I remove the `true` (which was just me testing originally following the docs) so I have this instead (below) then it still doesn't include the file which is present in that path. I don't see any errors being logged saying it can't find the file, the page is just blank.

```js
include: {
          resolvePath: (file) => {
            if (file.startsWith("@orgdocs")) {
              return file.replace("@orgdocs",
                path.resolve(__dirname, "../../node_modules/.github/"));
            }
            return file;
          },
        },
```

> The alias no longer ends with `.js`, check the docs, some webpack loader would check the extension, and assigning a js file to another file as alias can be problematic in some cases.

Sorry, which issue does this one refer to?

**Reply by Mister-Hope** [reply]
Some users report them in social group, you can just remove the .js prefix and everything should work again.

**Reply by Daeraxa** [reply]
I'm really sorry but I don't understand, what .js prefix and where, what issue does this resolve? The includes?

**Reply by Mister-Hope** [reply]
The homepage one https://github.com/pulsar-edit/pulsar-edit.github.io/pull/144/files

**Reply by Daeraxa** [reply]
This from `theme/index.js`
```js
  alias: {
    // https://vuepress-theme-hope.github.io/v2/cookbook/advanced/replace.html#replace-components
    "@theme-hope/components/HomePage.js": path.resolve(
      __dirname,
      "../components/HomePage.vue"
    ),
```

Sorry I'm not familiar with Vuepress or Vue at all really so not really understanding exactly what has broken, why and where it should be fixed. Looking at the docs at https://theme-hope.vuejs.press/cookbook/advanced/replace.html#replace-components you don't have the `.js` after `HomePage`. 
However removing this just breaks the site here instead (which I guess is progress as it means the alias seems to be working now):

```
09:11:14 [vite] page reload /home/dae/vuepressupgrade/docs/.vuepress/components/HomePage.vue
Failed to resolve import "@theme-hope/components/HomeFeatures.js" from "docs/.vuepress/components/HomePage.vue". Does the file exist?
09:11:15 [vite] Internal server error: Failed to resolve import "@theme-hope/components/HomeFeatures.js" from "docs/.vuepress/components/HomePage.vue". Does the file exist?
  Plugin: vite:import-analysis
  File: /home/dae/vuepressupgrade/docs/.vuepress/components/HomePage.vue:2:31
  1  |  import HomeFeatures from '@theme-hope/components/HomeFeatures.js';
     |                   
```

And removing the `.js` from `HomeFeatures.js` here also doesn't work.

Edit - It looks like `HomeFeatures` is no longer a replaceable component in the listed `Theme Component Alias`? (https://theme-hope.vuejs.press/cookbook/advanced/replace.html#replace-components). I assume this been replaced with `@theme-hope/components/HomeFeatures.js`?

Removing `import HomeFeatures from '@theme-hope/components/HomeFeatures.js';` works (at the loss of the feature boxes) and I've tried replacing it with `import FeaturePanel from '@theme-hope/components/FeaturePanel';` but I'm then guessing it isn't displaying because this bit of the vue file is incorrect?
```vue
    <main class="home project" id="main-content" aria-labelledby="main-title">
        <HomeFeatures />
        <Content />
```

**Reply by Mister-Hope** [reply]
![image](https://user-images.githubusercontent.com/33315834/229068231-6baa9a9e-2192-4412-8bb8-a4555447422e.png)

**Reply by Mister-Hope** [reply]
This kind of internal change will not be marked in changelog, we are pretending you are an advanced user if you want to hack into our theme and make edits.

**Reply by Daeraxa** [reply]
No that's fair, you can probably guess that I didn't make this change, it was by another contributor but now we have to try to either maintain it or undo it...

I think I have an idea of what to do for this bit and if it comes to the worst it can be undone.

However the major thing that is still broken is the including of external files, do you have any idea what is wrong there?

**Reply by Mister-Hope** [reply]
Can you give me a reproduction? Demo and docs are working

**Reply by Daeraxa** [reply]
A good example would be the Privacy policy which you can find on `.../docs/resources/privacy/` or via Resources > Privacy Policy in the footer or docs sidebar.
It should be including the content from `node_modules/.github/PRIVACY_POLICY.md` - this works on the old version of vuepress/hope (ignore the live site, we have had issues with pnpm not downloading new files so it displays as "file not found" there).

**Comment by Mister-Hope** [comment]
Anyway, I just need to check https://github.com/pulsar-edit/pulsar-edit.github.io/pull/198 and ensure it's working right?

**Reply by Daeraxa** [reply]
That is the one with the upgrade, yes. I haven't committed any of the changes for stuff you have already kindly helped me with but the include problem should still be there.

**Reply by Mister-Hope** [reply]
Commit them, I will check that later

**Reply by Daeraxa** [reply]
Ok those changes have been committed. It should now show the custom homepage but with no feature boxes which is how I left that bit.

The include not working is where I mentioned before for the privacy policy:

> A good example would be the Privacy policy which you can find on .../docs/resources/privacy/ or via Resources > Privacy Policy in the footer or docs sidebar.
It should be including the content from node_modules/.github/PRIVACY_POLICY.md - this works on the old version of vuepress/hope (ignore the live site, we have had issues with pnpm not downloading new files so it displays as "file not found" there).

**Reply by Mister-Hope** [reply]
I open a pr for you. Clone it locally and see if it works well.

**Reply by Daeraxa** [reply]
I will do, thank you so much. I'll have to look at it a bit later on but I'll definitely let you know. Do you know exactly what the issue was with the `orgdocs` include?

**Reply by Mister-Hope** [reply]
It's working in my case, so you still do not see contents when cloning my branch?

**Reply by Daeraxa** [reply]
It was just curiosity as I had a quick look (I've been away from my actual computer today) and it looked pretty much the same as what we had (except it was in JS not TS). So was just wondering if you had found something wrong with the original implementation of it that stopped it from working once upgraded.

**Reply by Mister-Hope** [reply]
Nope, maybe deps problem, hard to find out something in a real project instead of demo.

I will update this pr soon today
